packages/component-base/src/overflow-controller.js

```diff
@@ -55,22 +54,26 @@ export class OverflowController {
 
     this.__resizeObserver.observe(this.host);
 
-    this.__childObserver = new FlattenedNodesObserver(this.host, (info) => {
-      info.addedNodes.forEach((node) => {
-        if (node.nodeType === Node.ELEMENT_NODE) {
-          this.__resizeObserver.observe(node);
-        }
-      });
-
-      info.removedNodes.forEach((node) => {
-        if (node.nodeType === Node.ELEMENT_NODE) {
-          this.__resizeObserver.unobserve(node);
-        }
+    this.__childObserver = new MutationObserver((mutations) => {
```

Comment:
Initial nodes also need to be observed with the resize observer. Also seems like a test for this is missing.

Reply:
Fixed. Added logic to observe initial child nodes. Also changed the test to wait for `nextRender` (so now it fails without a fix) and added a new test that covers observing resize of lazily added nodes.